internal/pkg/deploy/cloudformation/project_test.go

```diff
@@ -672,6 +672,78 @@ func TestAddAppToProject(t *testing.T) {
 	}
 }
 
+func TestRemoveAppFromProject(t *testing.T) {
+	mockProject := &archer.Project{
+		Name:      "testproject",
+		AccountID: "1234",
+	}
+
+	tests := map[string]struct {
+		app string
+
+		mockDescribeStackSet          func(t *testing.T, in *cloudformation.DescribeStackSetInput) (*cloudformation.DescribeStackSetOutput, error)
+		mockUpdateStackSet            func(t *testing.T, in *cloudformation.UpdateStackSetInput) (*cloudformation.UpdateStackSetOutput, error)
+		mockDescribeStackSetOperation func(t *testing.T, in *cloudformation.DescribeStackSetOperationInput) (*cloudformation.DescribeStackSetOperationOutput, error)
+
+		want error
+	}{
+		"should remove input app from the stack set": {
+			app: "test",
+			// Given there hasn't been a StackSet update - the metadata in the stack body will be empty.
+			mockDescribeStackSet: func(t *testing.T, in *cloudformation.DescribeStackSetInput) (*cloudformation.DescribeStackSetOutput, error) {
+				body, err := yaml.Marshal(stack.DeployedProjectMetadata{Metadata: stack.ProjectResourcesConfig{
+					Apps:    []string{"test", "firsttest"},
+					Version: 1,
+				}})
+				require.NoError(t, err)
+				return &cloudformation.DescribeStackSetOutput{
+					StackSet: &cloudformation.StackSet{
+						TemplateBody: aws.String(string(body)),
+					},
+				}, nil
+			},
+			mockUpdateStackSet: func(t *testing.T, in *cloudformation.UpdateStackSetInput) (*cloudformation.UpdateStackSetOutput, error) {
+				require.NotZero(t, *in.TemplateBody, "TemplateBody should not be empty")
+				configToDeploy, err := stack.ProjectConfigFrom(in.TemplateBody)
+				require.NoError(t, err)
+				require.ElementsMatch(t, []string{"firsttest"}, configToDeploy.Apps)
+				require.Empty(t, configToDeploy.Accounts, "There should be no new apps to deploy")
```

Comment:
nit: we are deleting an app but the message "There should be no new apps to deploy" is a bit confusing